src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -1738,15 +1765,11 @@ void QueryAnalyzer::evaluateScalarSubqueryIfNeeded(QueryTreeNodePtr & node, size
     }
 
     if (tmp_block.rows() != 0)
-        throw Exception(ErrorCodes::INCORRECT_RESULT_OF_SCALAR_SUBQUERY, "Scalar subquery returned more than one row");
+        throw Exception(ErrorCodes::INCORRECT_RESULT_OF_SCALAR_SUBQUERY, "Scalar subquery returned more than zero row");
```

Comment:
The exception was right. We check this `tmp_block` after subquery already returned one row.

Reply:
Thanks for the clarification. Reverted it. Its quite confusing as the check is for `rows != 0`